LiteDB/Utils/Extensions/WaitHandleExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace LiteDB.Utils.Extensions
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal static class WaitHandleExtensions
+    {
+        public static async Task<bool> WaitAsync(this WaitHandle handle)
+        {
+            var tcs = new TaskCompletionSource<bool>();
```

Comment:
```suggestion
            var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Reply:
Fixed, but this option is not available in .net45

Reply:
Then use Task.Run for .net45

Reply:
I think we can leave the net framework as it's